test/e2e/set/nodeconfig/nodeconfig_optimizations.go

```diff
@@ -0,0 +1,264 @@
+		if err != nil && !apierrors.IsNotFound(err) {
+			framework.Failf("Can't delete NodeConfig %q: %v", naming.ObjRef(ncTemplate), err)
+		}
+		if !apierrors.IsNotFound(err) {
+			err = framework.WaitForObjectDeletion(context.Background(), f.DynamicAdminClient(), scyllav1alpha1.GroupVersion.WithResource("nodeconfigs"), ncTemplate.Namespace, ncTemplate.Name, nil)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+
+		framework.By("Deleting ResourceQuota %q, if it exists", naming.ObjRef(ncTemplate))
+		err = f.KubeAdminClient().CoreV1().ResourceQuotas(naming.ScyllaOperatorNodeTuningNamespace).Delete(context.Background(), resourceQuotaName, metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			framework.Failf("Can't delete ResourceQuota %q: %v", naming.ManualRef(naming.ScyllaOperatorNodeTuningNamespace, resourceQuotaName), err)
+		}
+		if !apierrors.IsNotFound(err) {
+			err = framework.WaitForObjectDeletion(context.Background(), f.DynamicAdminClient(), corev1.SchemeGroupVersion.WithResource("resourcequota"), naming.ScyllaOperatorNodeTuningNamespace, resourceQuotaName, nil)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		}
+	})
+
+	g.It("should create tuning resources and tune nodes", func() {
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		nc := ncTemplate.DeepCopy()
+
+		g.By("Creating a NodeConfig")
+		nc, err := f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs().Create(ctx, nc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the NodeConfig to deploy")
+		ctx1, ctx1Cancel := context.WithTimeout(ctx, nodeConfigRolloutTimeout)
+		defer ctx1Cancel()
+		nc, err = utils.WaitForNodeConfigState(ctx1, f.ScyllaAdminClient().ScyllaV1alpha1().NodeConfigs(), nc.Name, utils.WaitForStateOptions{TolerateDelete: false}, utils.IsNodeConfigRolledOut, utils.IsNodeConfigDoneWithNodeTuningFunc(matchingNodes))
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyNodeConfig(ctx, f.KubeAdminClient(), nc)
+
+		// There should be a tuning job for every scylla node.
+		nodeJobList, err := f.KubeAdminClient().BatchV1().Jobs(naming.ScyllaOperatorNodeTuningNamespace).List(ctx, metav1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set{
+				naming.NodeConfigNameLabel:    nc.Name,
+				naming.NodeConfigJobTypeLabel: string(naming.NodeConfigJobTypeNode),
+			}).String(),
+		})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(nodeJobList.Items).To(o.HaveLen(len(matchingNodes)))
+
+		var jobNodeNames []string
+		for _, j := range nodeJobList.Items {
+			o.Expect(j.Annotations).NotTo(o.BeNil())
+			nodeName, found := j.Annotations[naming.NodeConfigJobForNodeKey]
+			o.Expect(found).To(o.BeTrue())
+
+			o.Expect(nodeName).NotTo(o.BeEmpty())
+			jobNodeNames = append(jobNodeNames, nodeName)
+		}
+
+		var matchingNodeNames []string
+		for _, node := range matchingNodes {
+			matchingNodeNames = append(matchingNodeNames, node.Name)
+		}
+	})
```

Comment:
Deploy guaranteed cluster and check if another job is created for Scylla container

Reply:
I've reacted to the original comment and removed the guaranteed test. We can extend it in a followup.